### PR TITLE
Polish Javadoc and API surface

### PR DESCRIPTION
**Pull request by donat**
Fixes https://github.com/gradle/gradle-private/issues/3954

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test QFL

File: subprojects/model-core/src/main/java/org/gradle/internal/reflect/validation/TypeValidationProblemRenderer.java [hunk 1]
```diff
@@ -122,7 +120,7 @@ public static String introductionFor(Map<String, String> additionalMetadata) {
         String rootType = ofNullable(additionalMetadata.get(TYPE_NAME))
             .filter(TypeValidationProblemRenderer::shouldRenderType)
             .orElse(null);
-        ProblemsPluginId pluginId = ofNullable(additionalMetadata.get(PLUGIN_ID)).map(DefaultPluginId::new).orElse(null);
+        DefaultPluginId pluginId = ofNullable(additionalMetadata.get(PLUGIN_ID)).map(DefaultPluginId::new).orElse(null);
```
Comment (donat) [inline subprojects/model-core/src/main/java/org/gradle/internal/reflect/validation/TypeValidationProblemRenderer.java:123]: `ProblemsPluginId` was part of the API surface but was only used in the type validation renderer. I removed it from the API and inlined the classes here.

**Comment by donat** [comment]
@bot-gradle test RFM

**Comment by donat** [comment]
@bot-gradle test RFM

File: subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilder.java [hunk 2]
```diff
@@ -51,17 +40,52 @@
 @Incubating
 public interface ProblemBuilder {
 
+    /**
+     * The long description of this problem.
+     *
+     * @param details the details
+     *
+     * @return this
```
Comment (hegyibalint) [inline subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilder.java:48]: 🤔 I'm not sure the `@return` entries are required here, especially if `this` is the only comment.
I would rather change it to something more descriptive, or delete it.

Comment (hegyibalint) [reply subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilder.java:48]: E.g.,
```suggestion
     * @return the same instance the method was called on
```

File: subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilderDefiningDocumentation.java [hunk 3]
```diff
@@ -19,13 +19,24 @@
 import org.gradle.api.Incubating;
 
 /**
- * Builder that is part of the {@link ProblemBuilder} concept.
+ * Builder for {@link Problem} instances requiring the specification of related documentation.
```
Comment (hegyibalint) [inline subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilderDefiningDocumentation.java:22]: 🤔 IMO this is a bit hard to read, my bid is something like this:
```suggestion
 * {@link Problem} instance builder requiring the specification of documentation.
```

Comment (hegyibalint) [reply subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilderDefiningDocumentation.java:22]: This OFC applies to all the builder javadocs

**Review by hegyibalint — CHANGES_REQUESTED** [review]
Left some nitpicking 😄 
Handle this as a soft approval; I think after you check the comments, you can merge immediately.

File: subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilder.java (hunk 2 above)
Comment (donat) [reply subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilder.java:48]: I have to disagree here. We use `* @return this` regularly in the Gradle javadoc.

File: subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilderDefiningDocumentation.java (hunk 3 above)
Comment (donat) [reply subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilderDefiningDocumentation.java:22]: 👍

**Comment by donat** [comment]
@bot-gradle test and merge

**Pull request merged**